src/EnergyPlus/HybridEvapCoolingModel.cc

```diff
@@ -0,0 +1,1591 @@
+			Minimum_Return_Air_Relative_Humidity = min;
+			Maximum_Return_Air_Relative_Humidity = max;
+			return true;
+		}
+
+
+		bool CMode::InitializeOSAFConstraints(double minOSAF, double maxOSAF)
+		{
+			//minimum 0.00, maximum 1.00, Outside air fractions below this value will not be considered.
+			//If this field is blank, the lower constraint on outside air fraction will be 0.00,default 0.10
+			Min_OAF = minOSAF;
+			Max_OAF = maxOSAF;
+			return true;
+		}
+		bool CMode::InitializeMsaRatioConstraints(double minMsa, double maxMsa)
+		{
+			//minimum 0.00, maximum 1.00, Supply air mass flow rate ratios below this value will not be considered.
+			//Supply air mass flow rate ratio describes supply air mass flow rate as a fraction of mass flow rate associated with the value in field : "System Maximum Supply Air Flow Rate".
+			//If this field is blank, the lower constraint on outside air fraction will be 0.00,default 0.10
+			Min_Msa = minMsa;
+			Max_Msa = maxMsa;
+			return true;
+		}
+		bool CMode::ValidPointer(int curve_pointer) {
+			if (curve_pointer >= 0) return true;
+			else  return false;
+		}
+		double CMode::CalculateCurveVal(double X_0, double X_1, double X_2, double X_3, double X_4, double X_5, double X_6, int mode_number, int curveType)
+		{
+			double Y_val = 0;
+
+			switch (curveType)
+			{
+			case TEMP_CURVE:
+				if (ValidPointer(Tsa_curve_pointer))
+				{
+					Y_val = NormalizationReference*CurveValue(Tsa_curve_pointer, X_1, X_2, X_3, X_4, X_5, X_6);
+				}
+				else {
+					Y_val = X_3;//return air temp
+				}
+				break;
+			case W_CURVE:
+
+				if (ValidPointer(HRsa_curve_pointer))
+				{
+					Y_val = NormalizationReference*CurveValue(HRsa_curve_pointer, X_1, X_2, X_3, X_4, X_5, X_6);//0.01;//
+				}
+				else {
+					Y_val = X_4;//return HR
+				}
+				break;
+			case POWER_CURVE:
+
+				if (ValidPointer(Psa_curve_pointer))
+				{
+					Y_val = Correction*CurveValue(Psa_curve_pointer, X_1, X_2, X_3, X_4, X_5, X_6);
+				}
+				else {
+					Y_val = 0;//or set a more reasonable default
+				}
+				break;
+			default:
+				break;
+			}
+
+			return Y_val;
+		}
+		void CMode::InitializeCurve(int curveType, int curve_ID)
+		{
+
+			switch (curveType)
+			{
+			case TEMP_CURVE:
+				Tsa_curve_pointer = curve_ID;
+				break;
+			case W_CURVE:
+				HRsa_curve_pointer = curve_ID;
+				break;
+			case POWER_CURVE:
+				Psa_curve_pointer = curve_ID;
+				break;
+			default:
+				break;
+			}
+		}
+
+		bool CMode::GenerateSolutionSpace(double ResolutionMsa, double ResolutionOSA)
+		{
+			double deltaMsa = Max_Msa - Min_Msa;
+			double deltaOAF = Max_OAF - Min_OAF;
+			if (deltaMsa < ResolutionMsa) deltaMsa = ResolutionMsa;
+			if (deltaOAF < ResolutionOSA) deltaOAF = ResolutionOSA;
+			double Msastep_size = (deltaMsa * ResolutionMsa);
+			double OAFsteps_size = (deltaOAF * ResolutionOSA);
+
+
+			for (double Msa_val = Max_Msa; Msa_val >= Min_Msa; Msa_val = Msa_val - Msastep_size)
```

Comment:
I don't understand this comment, do you mean you want the padding in the parentheses to to be tabs?